source/CScriptEngine.cpp

```diff
@@ -1,1038 +1,1073 @@
+            cs->SetActive(false);
+        });
+    }
+
+    void CScriptEngine::ReregisterAllScripts()
+    {
+        TRACE("Reregistering all custom scripts");
+        std::for_each(CustomScripts.begin(), CustomScripts.end(), [this](CCustomScript *cs) {
+            AddScriptToQueue(cs, activeThreadQueue);
+            cs->SetActive(true);
+        });
+    }
+
+    CCustomScript::CCustomScript(const char *szFileName, bool bIsMiss, CCustomScript *parent, int label)
+        : CRunningScript(), bSaveEnabled(false), bOK(false),
+        LastSearchPed(0), LastSearchCar(0), LastSearchObj(0),
+        CompatVer(CLEO_VERSION)
+    {
+        IsCustom(1);
+        bIsMission = bUseMissionCleanup = bIsMiss;
+        UseTextCommands = 0;
+        NumDraws = 0;
+        NumTexts = 0;
+
+        TRACE("Loading custom script %s...", szFileName);
+
+        try
+        {
+			std::ifstream is;
+			if (label != 0) // Create external from label.
+			{
+				if (!parent)
+					throw std::logic_error("Trying to create external thread from label without parent thread");
+				BaseIP = parent->GetBasePointer();
+				CurrentIP = parent->GetBasePointer() - label;
+				memcpy(Name, parent->Name, sizeof(Name));
+				dwChecksum = parent->dwChecksum;
+				parentThread = parent;
+				parent->childThreads.push_back(this);
+			}
+			else
+			{
+				using std::ios;
+				std::ifstream is(szFileName, std::ios::binary);
+				is.exceptions(std::ios::badbit | std::ios::failbit);
+				std::size_t length;
+				is.seekg(0, std::ios::end);
+				length = is.tellg();
+				is.seekg(0, std::ios::beg);
+
+				if (bIsMiss)
+				{
+					if (*MissionLoaded)
+						throw std::logic_error("Starting of custom mission when other mission loaded");
+					*MissionLoaded = 1;
+					BaseIP = CurrentIP = missionBlock;
+				}
+				else {
+					BaseIP = CurrentIP = new BYTE[length];
+				}
+				is.read(reinterpret_cast<char *>(BaseIP), length);
+
+				auto fname = strrchr(szFileName, '\\') + 1;
+				if (!fname) fname = strrchr(szFileName, '/') + 1;
+				if (fname < szFileName) fname = szFileName;
+				memcpy(Name, fname, sizeof(Name));
+				Name[7] = '\0';
+				dwChecksum = crc32(reinterpret_cast<BYTE *>(BaseIP), length);
+			}
+			lastScriptCreated = this;
+            bOK = true;
+        }
+        catch (std::exception& e)
+        {
+            TRACE("Error during loading of custom script %s occured.\nError message: %s", szFileName, e.what());
+        }
+        catch (...)
+        {
+            TRACE("Unknown error during loading of custom script %s occured.", szFileName);
+        }
+    }
+
+    CCustomScript::~CCustomScript()
+    {
+        if (BaseIP && !bIsMission) delete[] BaseIP;
+		RunScriptDeleteDelegate(reinterpret_cast<CRunningScript*>(this));
+		if (lastScriptCreated == this) lastScriptCreated = nullptr;
+    }
+
+	float VectorSqrMagnitude(CVector vector) { return vector.x * vector.x + vector.y * vector.y + vector.z * vector.z; }

```

Comment:
What is this for and why it is there?

Reply:
Used for find opcodes (0AE1, 0AE2, 0AE3). Just a little performance improvement for distance checking, pretty much a must in game development.
https://docs.unity3d.com/ScriptReference/Vector3-sqrMagnitude.html

Reply:
I see, thanks for the link